### PR TITLE
Adding Constructors to generated classes.

### PR DESCRIPTION
**Pull request by AnishaSrivastava1411**
Working to add constructors.

### CODE REVIEW

**Comment by AnishaSrivastava1411** [comment]
@goneall @rtgdk 
As I was working on the second constructor i.e., ```Constructor(IModelStore modelStore, String documentUri, String id,@Nullable ModelCopyManager copyManager, boolean create) ``` I could only add the first three parameters in the declaration and later it started giving error, though I followed the same mechanism to do.
Error is:-
```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.inject.internal.cglib.core.$ReflectUtils$1 (file:/usr/share/maven/lib/guice.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
WARNING: Please consider reporting this to the maintainers of com.google.inject.internal.cglib.core.$ReflectUtils$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[INFO] Scanning for projects...
[INFO] 
[INFO] -------------------< schema-to-java:schema-to-java >--------------------
[INFO] Building schema-to-java 0.0.1-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-jibx-plugin:1.3.1:schema-codegen (default-cli) @ schema-to-java ---
[INFO] Generating Java sources in src/main/source-code from schemas available in src/main/config...
Loaded and validated 1 specified schema(s)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.096 s
[INFO] Finished at: 2020-08-11T02:36:07+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.jibx:maven-jibx-plugin:1.3.1:schema-codegen (default-cli) on project schema-to-java: null: MojoExecutionException: IllegalArgumentException -> [Help 1]
[ERROR] 

```
May you please help in understanding the reason for this build failure?
Please check the code here: ```side-project/Decorator/org/spdx/jibx/SPDXClassDecorator.java```

**Comment by goneall** [comment]
@AnishaSrivastava1411 Run maven with the -X option and post the more detailed errors.

**Comment by AnishaSrivastava1411** [comment]
> @AnishaSrivastava1411 Run maven with the -X option and post the more detailed errors.

Here it is:-
```
[INFO] Generating Java sources in src/main/source-code from schemas available in src/main/config...
Loaded and validated 1 specified schema(s)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.679 s
[INFO] Finished at: 2020-08-11T10:42:10+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.jibx:maven-jibx-plugin:1.3.1:schema-codegen (default-cli) on project schema-to-java: null: MojoExecutionException: IllegalArgumentException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.jibx:maven-jibx-plugin:1.3.1:schema-codegen (default-cli) on project schema-to-java: null
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException
    at org.jibx.maven.AbstractCodeGenMojo.execute (AbstractCodeGenMojo.java:274)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.lang.IllegalArgumentException
    at org.eclipse.jdt.core.dom.SimpleName.setIdentifier (SimpleName.java:187)
    at org.eclipse.jdt.core.dom.AST.newSimpleName (AST.java:1303)
    at org.eclipse.jdt.core.dom.AST.newName (AST.java:1423)
    at org.spdx.jibx.SPDXClassDecorator.start (SPDXClassDecorator.java:156)
    at org.jibx.schema.codegen.ClassHolder.initClass (ClassHolder.java:668)
    at org.jibx.schema.codegen.StructureClassHolder.generate (StructureClassHolder.java:2111)
    at org.jibx.schema.codegen.PackageHolder.generate (PackageHolder.java:210)
    at org.jibx.schema.codegen.PackageHolder.generate (PackageHolder.java:227)
    at org.jibx.schema.codegen.CodeGen.buildDataModel (CodeGen.java:1572)
    at org.jibx.schema.codegen.CodeGen.generate (CodeGen.java:1781)
    at org.jibx.schema.codegen.CodeGen.main (CodeGen.java:2206)
    at org.jibx.maven.AbstractCodeGenMojo.execute (AbstractCodeGenMojo.java:271)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

```

**Comment by AnishaSrivastava1411** [comment]
@goneall @rtgdk Please review.

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,4 @@
 schema-to-java.eml
 schema-to-java.userlibraries
 /target/
+
```
Comment (rtgdk) [inline .gitignore:9]: Remove this file change from PR.

File: side-project/Decorator/org/spdx/jibx/SPDXClassDecorator.java [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+		     block2.statements().add(sci2);
+		   
+		     holder.addMethod(constr);  
+		     holder.addMethod(constr2);   			    
+    	 }
+
+		
+		if( holder.getSuperClassName()==null &&  matchName(holder.getName()) && !(holder instanceof EnumerationClassHolder) )  {    		
+	        holder.setSuperClassName(m_baseClass);
+		}
```
Comment (rtgdk) [inline side-project/Decorator/org/spdx/jibx/SPDXClassDecorator.java:167]: Keep the indentation same. A lot of code above doesn't adhere to the same indentation. Although, java is not indentation based language, still follow the conventional code styles.

File: src/main/custom/GlobalCustomization.xml [hunk 3]
```diff
@@ -1,3 +1,3 @@
-<schema prefer-inline="true" show-schema="false" import-docs="true" use-inner="true">
+<schema prefer-inline ="true" show-schema="false" import-docs ="true" use-inner="true">
```
Comment (rtgdk) [inline src/main/custom/GlobalCustomization.xml:1]: Not sure why this change is in the PR. Please keep the whitespace change in PR to minimum to keep the PR small.

File: .gitignore (hunk 1 above)
Comment (AnishaSrivastava1411) [reply .gitignore:9]: sir, these are already in .gitignore. How I should remove it from PR?

Comment (AnishaSrivastava1411) [reply .gitignore:9]: Also, this creates conflict each time I commit changes.

**Comment by AnishaSrivastava1411** [comment]
@rtgdk , @goneall .I am trying to align codes in the same indentation, but after committing it gets changed. How to resolve this?

**Comment by rtgdk** [comment]
@AnishaSrivastava1411 I can see force-push of commits and merging of master frequently.. Not sure what steps you are following. You just need to make changes in your local branch of "AnishaSrivastava1411:master" (check your origin url - `git remote -v,`), commit the local changes, and push it to your origin (in this case your repo's master). Don't pull master (neither your repo or spdx repo) and don't force push.

**Comment by rtgdk** [comment]
@AnishaSrivastava1411  Can you provide the steps you follow while pushing some change (from git add .... to git push)

**Comment by rtgdk** [comment]
@AnishaSrivastava1411  you edited my comment, instead of replying 😆 
Don't force push ever, this might remove some previous uncommitted work of others/yours.
Use these commands

> git add .
git commit -m "Message"
git push origin master

**Comment by AnishaSrivastava1411** [comment]
> @AnishaSrivastava1411 you edited my comment, instead of replying 
> Don't force push ever, this might remove some previous uncommitted work of others/yours.
> Use these commands
> 
> > git add .
> > git commit -m "Message"
> > git push origin master

oops...🙈 :stuck_out_tongue:

will follow these commands now onwards.
